### PR TITLE
Working JSON Save system

### PR DESCRIPTION
**Pull request by DMP9-MC**
JSON Save system concept using bukkit's built-in GSON. Fully tested and
working.

### CODE REVIEW

File: src/me/sirfaizdat/prison/mines/MinesManager.java [hunk 1]
```diff
@@ -71,7 +74,34 @@ private void broadcastToWorld(String s, World w) {
     }
 
     public void load() {
-        for (File file : getAllMineFiles()) {
+    	boolean convert = Prison.i().config.useJson;
+    	if (!convert){Prison.l.warning("The new mine format converter is disabled");Prison.l.warning("Set the json option to true in the config.yml to enable it");}
```
Comment (SirFaizdat) [inline src/me/sirfaizdat/prison/mines/MinesManager.java]: Perhaps it should be enabled by default?

**Comment by DMP9-MC** [comment]
Will modify to implement changes, thanks.

File: src/me/sirfaizdat/prison/mines/MinesManager.java (hunk 1 above)
Comment (DMP9-MC) [inline src/me/sirfaizdat/prison/mines/MinesManager.java]: Part of me agrees, then again there will probably be somebody out there who will want to keep the old system. I say keep it (for now) for backwards compatibility.

**Comment by DMP9-MC** [comment]
Implemented changes, take a look at it. I believe I've got rid of all the Prison GUI remnants and all the eclipse files are gone.

**Comment by DMP9-MC** [comment]
*bump*

File: src/plugin.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
-name: ${project.name}
-version: ${project.version}
-description: ${project.description}
+name: Prison
```
Comment (SirFaizdat) [inline src/plugin.yml]: The name, version, and description values for `plugin.yml` are automatically set upon compilation by Maven.

File: src/me/sirfaizdat/prison/core/Config.java [hunk 3]
```diff
@@ -78,6 +78,18 @@ private void loadAll() {
             enableMultiworld = c.getBoolean("multiworld");
             fireworksOnRankup = c.getBoolean("fireworks-on-rankup");
             flamesOnAutosmelt = c.getBoolean("flames-on-autosmelt");
+            boolean configerr = false;
```
Comment (SirFaizdat) [inline src/me/sirfaizdat/prison/core/Config.java]: Slight nitpick here, but why is this variable named `configerr`? Perhaps a more descriptive/clear name could be used, or maybe I am misunderstanding something.

Comment (DMP9-MC) [inline src/me/sirfaizdat/prison/core/Config.java]: **Config**uration**Err**or

Will change to configmissing

File: src/plugin.yml (hunk 2 above)
Comment (DMP9-MC) [inline src/plugin.yml]: Ohh that's from when I was testing! Will change, thanks.

**Comment by SirFaizdat** [comment]
After those changes are made, do you think this is ready for merge?

**Comment by DMP9-MC** [comment]
I can test before merging, but other than that yeah

**Comment by SirFaizdat** [comment]
Alright, once you finish testing and making any changes, I'll merge.

**Comment by DMP9-MC** [comment]
Can you build the latest commit, please? My JDK isn't working...

**Comment by DMP9-MC** [comment]
Done testing! Sorry it took so long. You can merge now :)

**Pull request merged**